### PR TITLE
Lk1757

### PR DESCRIPTION
**Pull request by reish981**

### CODE REVIEW

File: src/test/java/edu/ksu/canvas/attendance/repository/AttendanceRepositoryImplUTest.java [hunk 1]
```diff
@@ -42,8 +42,9 @@ public void saveInBatches_tollerateNullAttendances() {
     public void deleteAttendanceByCourseAndDayOfClass_NullDateOfClass() {
```
Comment (KSU-Github) [inline src/test/java/edu/ksu/canvas/attendance/repository/AttendanceRepositoryImplUTest.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS00100)

File: src/main/java/edu/ksu/canvas/attendance/repository/AttendanceRepositoryImpl.java [hunk 2]
```diff
@@ -97,19 +97,20 @@ public void saveInBatches(List<Attendance> attendances) {
...
                 "WHERE a.attendanceStudent.canvasCourseId = :courseId " +
                 "AND trunc(a.dateOfClass) = trunc(:dateOfClass)" +
-                ")";
+                "AND a.attendanceStudent.canvasSectionId = :sectionId)";
 
         javax.persistence.Query query = entityManager.createQuery(jpql);
         query.setParameter("courseId", courseId);
         query.setParameter("dateOfClass", dateOfClass, TemporalType.DATE);
+        query.setParameter("sectionId", sectionId);
         int result = query.executeUpdate();
```
Comment (KSU-Github) [inline src/main/java/edu/ksu/canvas/attendance/repository/AttendanceRepositoryImpl.java]: MAJOR Remove this unused "result" local variable. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1481)

File: src/main/webapp/WEB-INF/jsp/roster.jsp [hunk 3]
```diff
@@ -122,26 +121,30 @@
                                 <input id="saveAttendanceOnTop" class="hovering-purple-button" type="submit" name="saveAttendance"
                                        value="Save Attendance"/>
                             </div>
-                            <div class="col-md-4 deleteAttendanceButton">
+                            <div class="col-md-3 deleteAttendanceButton">
                                 <a id="deleteAttendance" href="${context}/roster/${selectedSectionId}/delete" name="deleteAttendance" style="text-decoration: none" >
```
Comment (KSU-Github) [inline src/main/webapp/WEB-INF/jsp/roster.jsp]: MAJOR Remove this deprecated "name" attribute. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3AS1827)

**Comment by KSU-Github** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [roster.jsp#L9](https://github.com/kstateome/lti-attendance/blob/bdb744fa476588770790170c05355a2a77305ead/src/main/webapp/WEB-INF/jsp/roster.jsp#L9): Add a 'favicon' declaration in this 'header' tag. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3APageWithoutFaviconCheck)
1. ![MINOR][MINOR] [AttendanceRepositoryImplUTest.java#L26](https://github.com/kstateome/lti-attendance/blob/bdb744fa476588770790170c05355a2a77305ead/src/test/java/edu/ksu/canvas/attendance/repository/AttendanceRepositoryImplUTest.java#L26): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [AttendanceRepositoryImplUTest.java#L35](https://github.com/kstateome/lti-attendance/blob/bdb744fa476588770790170c05355a2a77305ead/src/test/java/edu/ksu/canvas/attendance/repository/AttendanceRepositoryImplUTest.java#L35): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS00100)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by kawesome** [comment]
Can one of the admins verify this patch?

File: src/main/java/edu/ksu/canvas/attendance/repository/AttendanceRepositoryImpl.java (hunk 2 above)
Comment (zoglmannk) [reply src/main/java/edu/ksu/canvas/attendance/repository/AttendanceRepositoryImpl.java]: This issue is minor. I concur with Sonar, and I have asked for additional changes in LK1764, so I'll kick this pull request back for a 30-second change.

**Comment by KSU-Github** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [roster.jsp#L9](https://github.com/kstateome/lti-attendance/blob/a9a889d3f17a19a4b934eac061f8d2b4f2422b4f/src/main/webapp/WEB-INF/jsp/roster.jsp#L9): Add a 'favicon' declaration in this 'header' tag. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3APageWithoutFaviconCheck)
1. ![MINOR][MINOR] [AttendanceRepositoryImplUTest.java#L26](https://github.com/kstateome/lti-attendance/blob/a9a889d3f17a19a4b934eac061f8d2b4f2422b4f/src/test/java/edu/ksu/canvas/attendance/repository/AttendanceRepositoryImplUTest.java#L26): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS00100)
1. ![MINOR][MINOR] [AttendanceRepositoryImplUTest.java#L35](https://github.com/kstateome/lti-attendance/blob/a9a889d3f17a19a4b934eac061f8d2b4f2422b4f/src/test/java/edu/ksu/canvas/attendance/repository/AttendanceRepositoryImplUTest.java#L35): Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS00100)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**